codegenerator/cli/src/config_parsing/system_config.rs

```diff
@@ -923,92 +893,93 @@ impl SyncSource {
+            Some(config) => Some(config.url.to_string()),
+            None => match hypersync_endpoints::get_default_hypersync_endpoint(network.id) {
+                Ok(url) => Some(url),
+                Err(_) => None,
+            },
+        };
+        let raw_rpcs = match (network.rpc_config, network.rpc) {
+            (Some(_), Some(_)) => Err(anyhow!("EE106: Cannot define both rpc and deprecated rpc_config for the same network, please only use the rpc option. Read more in our docs https://docs.envio.dev/docs/configuration-file"))?,
+            (None, Some(NetworkRpc::Url(url))) => vec![Rpc {
+                url: url.to_string(),
+                source_for: match hypersync_endpoint_url {
+                  Some(_) => For::Fallback,
+                  None => For::Sync,
+                },
+                sync_config: None,
+            }],
+            (None, Some(NetworkRpc::Single(rpc))) => vec![rpc],
+            (None, Some(NetworkRpc::List(list))) => list,
+            (Some(rpc_config), None) => match rpc_config.url {
+                SingleOrList::Single(url) => vec![Rpc {
+                    url: url.to_string(),
+                    source_for: For::Sync,
+                    sync_config: rpc_config.sync_config,
+                }],
+                SingleOrList::List(urls) => urls
+                    .iter()
+                    .map(|url| Rpc {
+                        url: url.to_string(),
+                        source_for: For::Sync,
+                        sync_config: rpc_config.sync_config.clone(),
+                    })
+                    .collect(),
+            },
+            (None, None) => vec![],
+        };
+
+        let mut rpcs = vec![];
+        for rpc in raw_rpcs.iter() {
+            match parse_url(rpc.url.as_str()) {
+              None => return Err(anyhow!("EE109: The RPC url \"{}\" is incorrect format. The RPC url needs to start with either http:// or https://", rpc.url)),
+              Some(url) => rpcs.push(Rpc {
+                  url,
+                  ..rpc.clone()
+              })
             }
-            human_config::evm::Network {
-              hypersync_config: None,
-              rpc_config: None,
-                ..
-            } => {
-                let defualt_hypersync_endpoint = hypersync_endpoints::get_default_hypersync_endpoint(network.id)
-                    .context("EE106: Undefined network config, please provide rpc_config, read more in our docs https://docs.envio.dev/docs/configuration-file")?;
-                Ok(Self::HypersyncConfig(HypersyncConfig {
-                    endpoint_url: defualt_hypersync_endpoint,
-                    is_client_decoder,
-                }))
+        }
+
+        let rpc_for_sync = rpcs.iter().find(|rpc| rpc.source_for == For::Sync);
+
+        let main = match rpc_for_sync {
+            Some(rpc) => {
+                match network.hypersync_config {
```

Comment:
I suggest running `cargo clippy` it should suggest using `if network.hypersync_config.is_some()`

Reply:
Done in the https://github.com/enviodev/hyperindex/pull/467